internal/server/http/handler.go

```diff
@@ -55,3 +72,148 @@ func fetchBodyFromFile(bodyFile string) (bytes []byte) {
 	}
 	return
 }
+
+func applyTemplate(i Imposter, bodyBytes []byte, r *http.Request) ([]byte, error) {
+	bodyStr := string(bodyBytes)
+
+	// check if the body contains a template
+	if !strings.Contains(bodyStr, "{{") {
+		return bodyBytes, nil
+	}
+
+	tmpl, err := template.New("body").
+		Funcs(template.FuncMap{
+			"stringsJoin": strings.Join,
+			"jsonMarshal": func(v interface{}) (string, error) {
```

Comment:
I'd prefer if we can have these functions declared somewhere else, with their own tests (at least the happy path), and just referenced here. Like with `strings.Join`, but for the custom ones.

We could have a package like: `internal/template` or `internal/templating` specifically for that.